src/utils.jl

```diff
@@ -180,5 +180,50 @@ makesym(t::Symbolic; kwargs...) = Sym{symtype(t)}(tosymbol(t; kwargs...))
 makesym(t::Num; kwargs...) = makesym(value(t); kwargs...)
 
 var_from_nested_derivative(x, i=0) = (missing, missing)
-var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1],i+1) : (x,i)
-var_from_nested_derivative(x::Sym,i=0) = (x,i)
+var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1], i + 1) : (x, i)
+var_from_nested_derivative(x::Sym,i=0) = (x, i)
+
+function degree(p::Sym, sym=nothing)
+    if sym === nothing
+        return 1
+    else
+        return Int(isequal(p, sym))
+    end
+end
+
+function degree(p::Pow, sym=nothing)
+    return p.exp * degree(p.base, sym)
+end
+
+function degree(p::Add, sym=nothing)
+    return maximum(degree(key, sym) for key in keys(p.dict))
+end
+        
+function degree(p::Mul, sym=nothing)
+    if sym === nothing
+        return sum(values(p.dict))
+    else
+        return get(p.dict, sym, 0)
+    end
+end
+
+function degree(p::Term, sym=nothing)
+    if sym === nothing
+        return 1
+    else
+        return Int(isequal(p, sym))
+    end
+end
+ 
+function degree(p, sym=nothing)
+    @assert !(typeof(p) <: Array) "Input may not be an Array."
+    p = value(p)
+    sym = value(sym)
+    if p isa Number
+        return 0
+    end
+    if isequal(p, sym)
+        return 1
+    end
+    return degree(p, sym)
```

Comment:
The check should be before this line.

Reply:
Yeah, I added it, will push a bit later.